src/logic/sanitize.ts

```diff
@@ -24,3 +24,20 @@ export const sanitize = (text: string) => {
   if (!safeSummary) throw new Error(`No content after sanitizing`);
   return safeSummary;
 };
+
+//TODO: move this to a more appropriate place
+export const validateUrl = async (url: string) => {
```

Comment:
In some cases this will be called with a nullish value, current. behaviour already handles that case correctly, so IMO worth expanding the signature to allow it
```suggestion
export const validateUrl = async (url: string | null | undefined) => {
```